Maddori.Apple/Maddori.Apple/Screen/InProgress/InProgressViewController.swift

```diff
@@ -0,0 +1,208 @@
+        }
+        
+        view.addSubview(subTitleLabel)
+        subTitleLabel.snp.makeConstraints {
+            $0.top.equalTo(titleLabel.snp.bottom).offset(SizeLiteral.titleSubtitleSpacing)
+            $0.leading.equalToSuperview().inset(SizeLiteral.leadingTrailingPadding)
+        }
+        
+        view.addSubview(keywordCollectionView)
+        keywordCollectionView.snp.makeConstraints {
+            $0.top.equalTo(subTitleLabel.snp.bottom)
+            $0.leading.trailing.bottom.equalToSuperview()
+        }
+    }
+    
+    override func setupNavigationBar() {
+        super.setupNavigationBar()
+        
+        let button = removeBarButtonItemOffset(with: backButton, offsetX: 10)
+        let backButton = makeBarButtonItem(with: button)
+        
+        navigationController?.navigationBar.prefersLargeTitles = false
+        navigationItem.largeTitleDisplayMode = .never
+        navigationItem.leftBarButtonItem = backButton
+    }
+    
+    private func setUpDelegation() {
+        keywordCollectionView.delegate = self
+        keywordCollectionView.dataSource = self
+    }
+    
+    private func setUpKeywordType() {
+        if isUserRetrospective {
+            for i in 0..<keywordData.count {
+                keywordData[i].type = .defaultKeyword
+            }
+            keywordsSectionList.append(keywordData)
+        } else {
+            var myKeywords = keywordData.filter { $0.from == user}
+            var otherKeywords = keywordData.filter { $0.from != user}
+            for i in 0..<myKeywords.count {
+                myKeywords[i].type = .defaultKeyword
+            }
+            for j in 0..<otherKeywords.count {
+                otherKeywords[j].type = .subKeyword
+            }
+            keywordsSectionList.append(myKeywords)
+            keywordsSectionList.append(otherKeywords)
+        }
+    }
+}
+
+// MARK: - extension
+
+extension InProgressViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return keywordsSectionList[0].count
+        case 1:
+            return keywordsSectionList[1].count
+        default:
+            return keywordsSectionList[0].count
+        }
+    }
+    
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard let cell = keywordCollectionView.dequeueReusableCell(withReuseIdentifier: KeywordCollectionViewCell.className, for: indexPath) as? KeywordCollectionViewCell else { return }
+        let section = indexPath.section
+        let item = indexPath.item
+        keywordsSectionList[section][item].type = .disabledKeyword
+        UIView.performWithoutAnimation {
+            cell.configLabel(type: .disabledKeyword)
+            cell.configShadow(type: .disabledKeyword)
+            self.keywordCollectionView.reloadItems(at: [indexPath])
+        }
```

Comment:
p4;

뭔가 여기서만 적용되는게 이상할지도 모르겠지만 메인쓰레드를 보장해줘야하는것 같은 느낌이 든단 말이죠..?? 깜빡거리는 UI를 해결하기 위한 코드라, UI에 직접적인 영향이 가는 것 같아요 ! 
```suggestion
        DispatchQueue.main.async {
            UIView.performWithoutAnimation {
                cell.configLabel(type: .disabledKeyword)
                cell.configShadow(type: .disabledKeyword)
                self.keywordCollectionView.reloadItems(at: [indexPath])
            }
        }
```
그럼 왜 여기만 메인 쓰레드를 보장하느냐, 다른 UI적 변화에도 다 보장해줘야하는 것 아니냐 ! 라고 하시면... 맞는 말씀입니다... 하하


혹시 틀렸을 수도 있으니 의견 부탁드립니다 !

Reply:
이게 확인할 수 있는 방법인지는 잘 모르겠지만 
configLabel과 configShadow가 실행되는 위치에서 어떤 쓰레드를 사용하는지 확인해봤습니다

<img width="550" alt="image" src="https://user-images.githubusercontent.com/72431640/199144800-1ac9a25c-957a-4333-bbb1-10fa1a40b6b8.png">
<img width="550" alt="image" src="https://user-images.githubusercontent.com/72431640/199144828-8aac4d9e-c28e-46e0-86f0-a19843cfc253.png">

cell을 눌렀을 때 어떤 쓰레드를 사용하는지 확인할 수 있다고 판단했는데 맞을까요?!
맞다면 메인 쓰레드를 잘 사용하는 것 같군요 🤔

Reply:
저희 텍스트필드에서 딸깍거리는 문제에서 쓰레드가 잘 보장이 되는지 확인 해봤는데요
<img width="561" alt="image" src="https://user-images.githubusercontent.com/78677571/199148520-03449935-bfcd-4584-82f5-245664b23fa5.png">

<img width="783" alt="image" src="https://user-images.githubusercontent.com/78677571/199148548-5a1228ac-0558-414c-9e94-9d595855b6fd.png">
global 쓰레드로 설정하고 프린트해봤을 때 이렇게 뜨는걸 보아하니 잘 되는 것 같습니다 !